great_expectations/core/expectation_suite.py

```diff
@@ -141,6 +143,56 @@ def expectations(self) -> list[Expectation]:
             for expectation_configuration in self.expectation_configurations
         ]
 
+    def add(self, expectation: Expectation) -> Expectation:
+        """Add an Expectation to the collection."""
+        if not any(
+            expectation.configuration == existing_config
+            for existing_config in self.expectation_configurations
+        ):
+            self.expectation_configurations.append(expectation.configuration)
+        else:
+            pass  # suite is a set-like collection
+        try:
+            self.save()
+        except Exception as exc:
+            # rollback this change
+            self.expectation_configurations.pop()
+            raise exc
+
+        return expectation
+
+    def delete(self, expectation: Expectation) -> Expectation:
+        """Delete an Expectation from the collection.
+
+        Raises:
+            ValueError: Expectation not found in suite.
+        """
+        updated_expectation_configs = [
+            exp_config
+            for exp_config in self.expectation_configurations
+            if exp_config != expectation.configuration
+        ]
+        if len(updated_expectation_configs) != len(self.expectation_configurations) - 1:
+            raise KeyError("No matching expectation was found.")
+        self.expectation_configurations = updated_expectation_configs
+
+        try:
+            self.save()
+        except Exception as exc:
+            # rollback this change
+            # expectation suite is set-like so order of expectations doesn't matter
```

Comment:
If it is set-like, could we have the internal representation actually be a set instead of a list?

Reply:
i decided against this approach since items in a set need to be hashable - we would have to serialize the expectations (or expectation configurations), and it just seems more convenient to store them in the state we want to work with them.